### ISSUE TITLE
Include various iteration examples

### ISSUE THREAD

**Issue by aishfenton** [japgolly/scalajs-benchmark#10]
Dirty secret of Scala collections is that while-loops are much faster than foreach, and for comprehensions. Would be interesting to see those in the benchmark.

**Comment by japgolly** [comment]
Good idea. A PR for this would be awesome. I'd be interested to see this.
